File: rails/spec/factories/equipment.rb
FactoryGirl.define do
  factory :equipment do
    sequence(:name) { |n| "equipment-name-#{n}" }
    sequence(:is_asset_tag) { |n| "asset-tag-#{n}" }
    sequence(:serial_number) { |n| "serial-number-#{n}" }
    sequence(:model_number) { |n| "model-number-#{n}" }
    sequence(:product_number) { |n| "product-number-#{n}" }
  end
end

File: ember/app/models/equipment.js
import DS from 'ember-data';

export default DS.Model.extend({
  name:          DS.attr('string'),
  isAssetTag:    DS.attr('string'),
  serialNumber:  DS.attr('string'),
  modelNumber:   DS.attr('string'),
  productNumber: DS.attr('string'),
  equipmentType: DS.belongsTo('equipment-type', {polymorphic: true, async: true})
});
